Storage/StorageTests/CoreData/MigrationTests.swift

```diff
@@ -305,6 +305,43 @@ final class MigrationTests: XCTestCase {
         XCTAssertNotNil(migratedOrder.entity.attributesByName["paymentMethodID"])
         XCTAssertNil(migratedOrder.value(forKey: "paymentMethodID"))
     }
+
+    func test_migrating_from_37_to_38_enables_creating_new_shipping_labels_entities() throws {
+        // Given
+        let sourceContainer = try startPersistentContainer("Model 37")
+        let sourceContext = sourceContainer.viewContext
+        try sourceContext.save()
+
+        // When
+        let targetContainer = try migrate(sourceContainer, to: "Model 38")
+
+        // Then
+        let targetContext = targetContainer.viewContext
+
+        XCTAssertEqual(try targetContext.count(entityName: "ShippingLabel"), 0)
+        XCTAssertEqual(try targetContext.count(entityName: "ShippingLabelAddress"), 0)
+        XCTAssertEqual(try targetContext.count(entityName: "ShippingLabelRefund"), 0)
+        XCTAssertEqual(try targetContext.count(entityName: "ShippingLabelSettings"), 0)
+
+        // Creates an `ShippingLabel` with all relationships.
+        let originAddress = insertShippingLabelAddress(to: targetContext)
+        let destinationAddress = insertShippingLabelAddress(to: targetContext)
+        let shippingLabelRefund = insertShippingLabelRefund(to: targetContext)
+        let shippingLabel = insertShippingLabel(to: targetContext)
+        shippingLabel.setValue(originAddress, forKey: "originAddress")
+        shippingLabel.setValue(destinationAddress, forKey: "destinationAddress")
+        shippingLabel.setValue(shippingLabelRefund, forKey: "refund")
+
+        // Creates an `ShippingLabelSettings`.
+        _ = insertShippingLabelSettings(to: targetContext)
+
+        try targetContext.save()
```

Comment:
You may wish to wrap this in `XCTAssertNoThrow` and remove the throws from the test method. I know the rest of the class doesn't follow this, but it's nice to make sure there aren't any unexpected throws and it looks nicer on execution of the tests if there is a failure.

Reply:
good suggestion, updated in https://github.com/woocommerce/woocommerce-ios/pull/3168/commits/5520ca5e34f23076f981fbfec044e97c8ba6b1f1! however, I wasn't able to remove the `throws` from this test case because there are other `try` statements that assign to a constant value and I couldn't find a good way to use `XCTAssertNoThrow` for these two:

- `let sourceContainer = try startPersistentContainer("Model 37")`
- `let targetContainer = try migrate(sourceContainer, to: "Model 38")`

do you have any ideas?

Reply:
What about?
```
let sourceContainer = XCTUnwrap(try? startPersistentContainer("Model 37"))
```

Not sure if that is better or worse 😅 
On one hand, the failures are easy to see but on the other, the code looks a bit odd.

Reply:
Ooh nice! I agree on the tradeoff, I think I slightly prefer code readability to how the test fails. What do you think @astralbodies?

Reply:
I don't believe we need to change the things like `let sourceContainer = try startPersistentContainer("Model 37")`. 

Using `XCTAssertNoThrow` _can_ make sense for `try targetContext.save()` if we are _proving_ that it will not fail. I think for this case, we're not. From what I can tell, the proofs that we truly want are the `XCTAssertEqual()` below.

So, in this case, I would prefer to have a less awkward and more readable option.